develop/CICD/Tutorials/CICD_Tutorial_GitHub_From_Code_To_Product.md

```diff
@@ -0,0 +1,136 @@
+
+   1. Select a name, specify your GitHub account, and mark yourself as the owner.
+
+   1. When you have entered all the information, click *Create and Push*.
+
+1. Go to the newly created GitHub repository on <https://github.com/>.
+
+   > [!TIP]
+   > In Visual Studio 2022, you can open the *GIT* menu and select *GitHub/View Pull Requests* to quickly access the correct repository.
+
+1. If your repository does not follow [Skyline's naming conventions](xref:Using_GitHub_for_CICD#repository-naming-convention), specify the artifact type:
+
+   1. Go to the *Code* tab.
+
+   1. Click the cogwheel next to *About*.
+
+   1. Under *Topics*, add a topic with the artifact type, as defined under [Inferring Catalog Item Type](https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.Tools.GitHubToCatalogYaml?tab=readme-ov-file#inferring-catalog-item-type).
+
+> [!TIP]
+> For more information, refer to the [GitHubToCatalogYaml readme](https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.Tools.GitHubToCatalogYaml#readme-body-tab).
+
+## Step 3: Add the Starter Workflow
+
+1. Go to the Starter Workflow:
+
+   - For **public** repositories or **paid** GitHub subscriptions:
+
+     On your GitHub page, go to the *Actions* tab and select the DataMiner workflow that matches your solution type. If none of the names match, select *DataMiner CICD AutomationScript*.
+
+   - For **private** repositories on a **free** GitHub subscription:
+
+     1. On your GitHub page, go to the *Actions* tab and select *Set up workflow yourself*.
+
+     1. Go to [Skyline-Starter-Workflows](https://github.com/SkylineCommunications/Skyline-Starter-Workflows/tree/main/workflow-templates) and select the DataMiner workflow that matches your solution type.
+
+        If none of the names match, select *DataMiner CICD AutomationScript*.
+
+     1. Copy the content of the workflow file and paste it into your new workflow file.
+
+1. Open a new browser window or tab, go to <https://sonarcloud.io/projects/create>, and create a project for the correct organization.
+
+1. Enter the ID of the project as shown in the SonarCloud project URL as the `sonarCloudProjectName`.
+
+1. Open a new browser window or tab, and create and add the necessary [GitHub secrets](xref:GitHub_Secrets) that you see in the workflow..
+
+1. Go back to your pipeline file and click *Commit Changes*.
+
+   Commit the changes directly to the master branch.
+
+1. Go to the *Actions* tab.
+
+   You should see your new workflow running. It will run for every Git push.
+
+You should now see your CI jobs complete successfully, ensuring that quality standards are met.
+
+## Step 4: Release a private item in the DataMiner Catalog
+
+1. On the *Code* tab for your repository on GitHub, on the right, click *Create a new release*.
+
+1. Click *Choose a tag* and enter the version, for example `1.0.0.1-beta1`.
+
+   You can use any version you want. For a pre-release, a "-text" suffix is usually used.
```

Comment:
Just to be sure, could you confirm that this is indeed the suffix used? It seems an odd choice.

Reply:
It can be -abc   -blabla   -helloworld   -beta1  ...
So I called it -text
Technically it's regex: -.*   but that didn't seem clear in a doc.

Reply:
```suggestion
   You can use any version you want. For a pre-release, usually a suffix is added, which can contain any text, e.g. "-beta1" in the example above.
```